airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/common.py

```diff
@@ -0,0 +1,155 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import json
+from abc import ABC, abstractmethod
+from http import HTTPStatus
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import requests
+from airbyte_cdk.sources.streams.core import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from pydantic import BaseModel, ValidationError
+from source_amazon_ads.common import PageToken, SourceContext
+from source_amazon_ads.schemas import JSModel
+
+URL_BASE = "https://advertising-api.amazon.com/"
+
+
+class ErrorResponse(BaseModel):
+    code: str
+    details: str
+    requestId: str
+
+
+class BasicAmazonAdsStream(Stream, ABC):
+    def __init__(self, config, context: SourceContext = None):
+        self._ctx = context or SourceContext()
+        self._config = config
```

Comment:
isn't it too generic, are we use all fields there?

Reply:
no, only client_id, updated